src/main/java/spireCafe/abstracts/AbstractMerchant.java

```diff
@@ -1,17 +1,55 @@
 package spireCafe.abstracts;
 
-import com.megacrit.cardcrawl.shop.ShopScreen;
+import basemod.BaseMod;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import spireCafe.screens.CafeMerchantScreen;
+
+import java.util.ArrayList;
 
 public abstract class AbstractMerchant extends AbstractCafeInteractable {
     public String name;
-    public ShopScreen shopScreen;
+    public TextureRegion background;
+    public ArrayList<AbstractArticle> articles = new ArrayList<>();
+    public ArrayList<AbstractArticle> toRemove = new ArrayList<>();
 
     public AbstractMerchant(float animationX, float animationY, float hb_w, float hb_h) {
         super(animationX, animationY, hb_w, hb_h);
-        shopScreen = createShopScreen();
     }
 
-    public abstract ShopScreen createShopScreen();
+    public abstract void rollShop();
+
+    @Override
+    public void onInteract() {
+        BaseMod.openCustomScreen(CafeMerchantScreen.ScreenEnum.CAFE_MERCHANT_SCREEN, this);
```

Comment:
Might be worth having a boolean to see if the shop was rolled and then calling rollShop in here in case it's false

Reply:
Example Merchant does not set the boolean to true in rollShop() implementation